hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/container/metrics/TestContainerMetrics.java

```diff
@@ -67,72 +84,126 @@
-          .createSingleNodePipeline();
-      OzoneConfiguration conf = new OzoneConfiguration();
-      conf.setInt(OzoneConfigKeys.HDDS_CONTAINER_IPC_PORT,
-          pipeline.getFirstNode()
-              .getPort(DatanodeDetails.Port.Name.STANDALONE).getValue());
-      conf.setInt(DFSConfigKeysLegacy.DFS_METRICS_PERCENTILES_INTERVALS_KEY,
-          interval);
-
-      DatanodeDetails datanodeDetails = randomDatanodeDetails();
-      conf.set(ScmConfigKeys.HDDS_DATANODE_DIR_KEY, path);
-      conf.set(OzoneConfigKeys.OZONE_METADATA_DIRS, path);
-      VolumeSet volumeSet = new MutableVolumeSet(
-          datanodeDetails.getUuidString(), conf,
-          null, StorageVolume.VolumeType.DATA_VOLUME, null);
-      ContainerSet containerSet = new ContainerSet(1000);
-      StateContext context = ContainerTestUtils.getMockContext(
-          datanodeDetails, conf);
-      ContainerMetrics metrics = ContainerMetrics.create(conf);
-      Map<ContainerProtos.ContainerType, Handler> handlers = Maps.newHashMap();
-      for (ContainerProtos.ContainerType containerType :
-          ContainerProtos.ContainerType.values()) {
-        handlers.put(containerType,
-            Handler.getHandlerForContainerType(containerType, conf,
-                context.getParent().getDatanodeDetails().getUuidString(),
-                containerSet, volumeSet, metrics,
-                c -> { }));
-      }
-      HddsDispatcher dispatcher = new HddsDispatcher(conf, containerSet,
-          volumeSet, handlers, context, metrics, null);
-      StorageVolumeUtil.getHddsVolumesList(volumeSet.getVolumesList())
-          .forEach(hddsVolume -> hddsVolume.setDbParentDir(tempDir.toFile()));
-      dispatcher.setClusterId(UUID.randomUUID().toString());
-
-      server = new XceiverServerGrpc(datanodeDetails, conf, dispatcher, null);
-      client = new XceiverClientGrpc(pipeline, conf);
+      final Pipeline pipeline =
+          MockPipeline.createSingleNodePipeline();
+      initConf.accept(pipeline);
 
+      DatanodeDetails dn = pipeline.getFirstNode();
+      volumeSet = createVolumeSet(dn, TEST_DIR + dn.getUuidString());
+      server = createServer.apply(dn, volumeSet);
       server.start();
+      initServer.accept(dn, pipeline);
+
+      client = createClient.apply(pipeline);
       client.connect();
 
       // Write Chunk
       BlockID blockID = ContainerTestHelper.getTestBlockID(containerID);
-      ContainerTestHelper.getWriteChunkRequest(
-          pipeline, blockID, 1024);
-      ContainerProtos.ContainerCommandRequestProto writeChunkRequest =
+      final ContainerProtos.ContainerCommandRequestProto writeChunkRequest =
           ContainerTestHelper.getWriteChunkRequest(
               pipeline, blockID, 1024);
-      ContainerCommandResponseProto response =
-              client.sendCommand(writeChunkRequest);
+      ContainerCommandResponseProto response = client.sendCommand(writeChunkRequest);
       assertEquals(ContainerProtos.Result.SUCCESS,
```

Comment:
we should check the related metrics in containerMetrics are correct. In specific, make sure numWriteChunk is expected.

Reply:
Here are some assertions for containerMetrics:
https://github.com/apache/ozone/pull/6835/files/fc89a7e43d2ea77f517ae03d09a16510448e44dd#diff-3d7dd1e263bbcdc0d201a231ebd91393746cd9ca653cff7de0bafc6b3047ae7dR214-L146
Is it enough?